.travis.yml

```diff
@@ -22,33 +22,39 @@ before_script:
 
 jobs:
   include:
-    - stage: Verify
+    - stage: Build
       script:
         - make install.tools
         - make .gitvalidation
         - make verify
+        - make binaries
       go: 1.8.x
     - script:
         - make install.tools
         - make .gitvalidation
         - make verify
+        - make binaries
       go: tip
-    - stage: Build and Test
+    - stage: Test
       script:
         - make install.deps
         - make test
-        - make binaries
         - make test-cri
+      after_script:
+        # Abuse travis to preserve the log.
+        - cat /tmp/test-cri/cri-containerd.log
+        - cat /tmp/test-cri/containerd.log
       go: 1.8.x
-    - script:
+    - stage: E2E Test
+      script:
+        # TODO(random-liu): Uncomment after test passes.
+        #- if [ "${TRAVIS_EVENT_TYPE}" != "cron" ]; then
+        #    echo "Skip node e2e test for non-cron job"
+        #    exit 0
+        #  fi
         - make install.deps
-        - make test
-        - make binaries
-        - make test-cri
-      go: tip
-
-after_script:
-    # Abuse travis to preserve the log.
-    # TODO(random-liu): Use prow for integration test.
-    - cat /tmp/cri-containerd.log
-    - cat /tmp/containerd.log
+        - make test-e2e-node
+      after_script:
+        # TODO(random-liu): Upload log to GCS.
+        - test -f /tmp/test-e2e-node/cri-containerd.log && cat /tmp/test-e2e-node/cri-containerd.log
```

Comment:
probably a todo here to figure out how to filter the logs a bit more so there's room in our 4m limit to get the containerd log...